### PR TITLE
Lep 334 remove assessment eligiblity

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/LEP-334

---

Remove all eligibility records from database - everything moved into CalculationOutput class. Eligibility Creator classes re-purposed to just make thresholds, as these are required by the results.

## Checklists

Author: (before you ask people to review this PR)

- [x] Diff - review it, ensuring it contains only expected changes
- [x] Whitespace changes - avoid if possible, because they make diffs harder to read and conflicts more likely
- [ ] Changelog - add a line, if it meets the criteria
- [x] Secrets - no secrets should be added
- [x] Commit messages - say *why* the change was made
- [x] PR description - summarizes *what* changed and *why*, with a JIRA ticket ID
- [x] Tests pass - on CircleCI
- [x] Conflicts - resolve if Github reports them. e.g. with `git rebase main`

Reviewers remember:

- [ ] Jira ticket criteria are met
- [ ] Migrations - test migration and rollback: `rake db:migrate && rake db:rollback`

### CODE REVIEW

File: app/services/decorators/v6/applicant_disposable_income_result_decorator.rb [hunk 1]
```diff
@@ -21,7 +22,10 @@ def as_json
     private
 
       def proceeding_types
```
Comment (ripan) [inline app/services/decorators/v6/applicant_disposable_income_result_decorator.rb:24]: @starswan What do you think of using some base result decorator to remove the duplication?

File: app/lib/gross_income/subtotals_base.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module GrossIncome
+  class SubtotalsBase
```
[the remark this answers is not in the document]
Comment (starswan) [reply app/lib/gross_income/subtotals_base.rb:2]: consistancy is good - clearly I couldn't decide what to call it.

File: app/models/eligibility/base.rb [hunk 3]
```diff
@@ -1,19 +1,11 @@
...
+    attribute :assessment_result, :string
 
-    def proceeding_type_code_validation
-      errors.add(:proceeding_type_code, "invalid: #{proceeding_type_code}") unless valid_proceeding_type_code?
-    end
-
-    def valid_proceeding_type_code?
-      proceeding_type_code.to_sym.in?(CFEConstants::VALID_PROCEEDING_TYPE_CCMS_CODES)
-    end
+    validates :assessment_result, inclusion: { in: %w[pending eligible ineligible contribution_required] }
```
[the remark this answers is not in the document]
Comment (starswan) [reply app/models/eligibility/base.rb:9]: This isn't my change - no idea why git thinks this is new.

File: app/models/eligibility/capital.rb [hunk 4]
```diff
@@ -1,23 +1,6 @@
...
-      if assessed_capital <= lower_threshold
-        :eligible
-      elsif assessed_capital <= upper_threshold
-        :contribution_required
-      else
-        :ineligible
-      end
-    end
+    attribute :lower_threshold, :decimal
+    attribute :upper_threshold, :decimal
```
[the remark this answers is not in the document]
Comment (starswan) [reply app/models/eligibility/capital.rb:4]: I'm not sure of the value - we're not saving them, and they are pretty weak.

File: app/services/decorators/v6/applicant_disposable_income_result_decorator.rb (hunk 1 above)
Comment (starswan) [reply app/services/decorators/v6/applicant_disposable_income_result_decorator.rb:24]: doesn't fit well with the LSP (Liskov Substitution Principal). Think maybe an eligibility should keep a reference to its own proceeding type though...

**Review by ripan — APPROVED** [review]

**Pull request merged**